### ISSUE TITLE
[Solved] BalenaAmbiguousApplication: Application is ambiguous error on balena os configure

### ISSUE THREAD

**Issue by vipulgupta2048** [balena-io/balena-cli#2189]
# Expected Behavior
An application called inkyshot deployed with DWB on balena-cloud.com
Using `balena os configure balena-cloud-raspberry-pi-2.54.2+rev1-dev-v11.12.4.img -a inkyshot` should configure the image for the application argument provided.

# Actual Behavior

Getting the error as below

```
➜  Desktop balena os configure balena-cloud-raspberry-pi-2.54.2+rev1-dev-v11.12.4.img -a inkyshot --debug
[debug] new argv=[/home/vipulgupta2048/.nvm/versions/node/v14.15.3/bin/node,/home/vipulgupta2048/.nvm/versions/node/v14.15.3/bin/balena,os:configure,balena-cloud-raspberry-pi-2.54.2+rev1-dev-v11.12.4.img,-a,inkyshot] length=6
BalenaAmbiguousApplication: Application is ambiguous: inkyshot

BalenaAmbiguousApplication: Application is ambiguous: inkyshot
    at Object.get (/home/vipulgupta2048/.nvm/versions/node/v14.15.3/lib/node_modules/balena-cli/node_modules/balena-sdk/es2018/models/application.js:257:27)
    at processTicksAndRejections (internal/process/task_queues.js:93:5)
    at async OsConfigureCmd.run (/home/vipulgupta2048/.nvm/versions/node/v14.15.3/lib/node_modules/balena-cli/build/commands/os/configure.js:39:20)
    at async OsConfigureCmd._run (/home/vipulgupta2048/.nvm/versions/node/v14.15.3/lib/node_modules/balena-cli/node_modules/@oclif/command/lib/command.js:43:20)
    at async Config.runCommand (/home/vipulgupta2048/.nvm/versions/node/v14.15.3/lib/node_modules/balena-cli/node_modules/@oclif/config/lib/config.js:175:24)
    at async CustomMain.run (/home/vipulgupta2048/.nvm/versions/node/v14.15.3/lib/node_modules/balena-cli/node_modules/@oclif/command/lib/main.js:27:9)
    at async CustomMain._run (/home/vipulgupta2048/.nvm/versions/node/v14.15.3/lib/node_modules/balena-cli/node_modules/@oclif/command/lib/command.js:43:20)
    at async Promise.all (index 1)
    at async oclifRun (/home/vipulgupta2048/.nvm/versions/node/v14.15.3/lib/node_modules/balena-cli/build/app.js:75:5)
    at async Object.run (/home/vipulgupta2048/.nvm/versions/node/v14.15.3/lib/node_modules/balena-cli/build/app.js:88:9)
```

I checked if I am in the right account and the application was visible using `balena apps`
I then checked my other applications (Both normally created and DWB) and only DWB applications couldn't be found. I tried balena-pihole and balena-nextcloud. 

# Steps to Reproduce the Problem

1. Use `Deploy with balena` to create an application
2. Try to configure a balenaOS image using `balena os configure IMG_NAME -a DWB_APP_NAME`
3. Get error

# Specifications

- **balena CLI version:** balena-cli version "12.36.1"
- **Cloud backend: openBalena or balenaCloud?** balenaCloud
- **Operating system version:** Ubuntu 20.04
- **32/64 bit OS and processor:** 64-bit Intel processor
- **Install method:** npm
- **If npm install, Node.js and npm version:** Node v14.15.3 and NPM v6.14.9

**Comment by vipulgupta2048** [comment]
Thought it might be a version thing since we updated our contracts to follow new spec, I am on CLI v 12.40.0 and still same error

**Comment by vipulgupta2048** [comment]
Found a solution. 

TLDR: Provide slug names as the team recommended in the documentation for [balena os configure](https://www.balena.io/docs/reference/balena-cli/#os-configure-image). 

The command should be `balena os configure IMG_NAME -a YOUR_ORG/APP_NAME`

I will paste it here for folks to read the reason if someone comes across it again. I think we should delete the other examples in the docs for balena os configure to make sure folks use app slugs. 

```
App slugs are the recommended option, as they are unique and unambiguous. Slugs can be listed with the balena apps command. Note that slugs may change if the application is renamed. App names are not unique and may result in "Application is ambiguous" errors at any time (even if it "used to work in the past"), for example, if the name clashes with a newly created public application, or with apps from other balena accounts that you may have been invited to as a member. For this reason, app names are especially discouraged in scripts (e.g. CI environments). 
```
